Sources/Core2/SubscriberConfigurator.cs

```diff
@@ -151,17 +165,50 @@ public ISubscriberConfigurator UseTrace(ITrace trace)
 
         public ISubscriberConfigurator UseDurableQueue(string queueName)
         {
+            _durable = true;
+
             _queueName = queueName;
 
+            _autoCreate = false;
+
             return this;
         }
 
         public ISubscriberConfigurator UseDurableQueue(string queueName, bool createBindings)
         {
+            _durable = true;
+
             _queueName = queueName;
 
             _createBindings = createBindings;
 
+            _autoCreate = false;
+
+            return this;
+        }
+
+        public ISubscriberConfigurator UseDurableQueue(string queueName, bool createBindings, bool autoCreate)
+        {
+            _durable = true;
+
+            _queueName = queueName;
+
+            _createBindings = createBindings;
+
+            _autoCreate = autoCreate;
+
+            return this;
+        }
+
+        public ISubscriberConfigurator UseNonDurableNamedQueue(string queueNameSuffix)
+        {
+            _durable = false;
+
+            _autoCreate = true;
+
+            _queueName = "serviceBus.gen-" + System.Environment.MachineName

```

Comment:
Looks like application specific naming convention. Please move it out of MB. 10x